### PR TITLE
Add special treatment of connection variables in SummaryState

### PR DESCRIPTION
**Pull request by joakim-hove**
The main usage of this class will be here: https://github.com/OPM/opm-common/pull/2373

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
Looks good to me.

File: src/opm/parser/eclipse/EclipseState/Schedule/SummaryState.cpp [hunk 1]
```diff
@@ -53,10 +53,11 @@ namespace {
...
     }
 
+    template <class T>
+    using map2 = std::unordered_map<std::string, std::unordered_map<std::string, T>>;
 
-    using map2 = std::unordered_map<std::string, std::unordered_map<std::string, double>>;
-
-    bool has_var(const map2& values, const std::string& var1, const std::string var2) {
+    template <class T>
+    bool has_var(const map2<T>& values, const std::string& var1, const std::string var2) {
```
[the remark this answers is not in the document]
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/SummaryState.cpp]: NO - should be reference; fixed

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**